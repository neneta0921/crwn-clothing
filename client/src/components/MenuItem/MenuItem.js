import { withRouter } from 'react-router-dom';

import {
  BackgroundImageContainer,
  MenuItemContainer,
  ContentContainer,
  ContentSubtitle,
  ContentTitle,
} from './MenuItemStyles';

export const MenuItem = ({ title, imageUrl, size, history, linkUrl, match }) => (
  <MenuItemContainer size={size} onClick={() => history.push(`${match.url}${linkUrl}`)}>
    <BackgroundImageContainer className="background-image" imageUrl={imageUrl} />
    <ContentContainer className="content">
      <ContentTitle>{title.toUpperCase()}</ContentTitle>
      <ContentSubtitle>SHOP NOW</ContentSubtitle>
    </ContentContainer>
  </MenuItemContainer>
);

export default withRouter(MenuItem);
