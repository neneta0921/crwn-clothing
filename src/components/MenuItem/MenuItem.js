import { withRouter } from 'react-router-dom';

import {
  BackgroundImageContainer,
  MenuItemContainer,
  ContentContainer,
  ContentSubtitle,
  ContentTitle
} from './MenuItem.styles';

const MenuItem = ({ title, imageUrl, size, history, linkUrl, match }) => (
  <MenuItemContainer size={size} onClick={() => history.push(`${match.url}${linkUrl}`)}>
    <BackgroundImageContainer className='background-image' imageUrl={imageUrl} />
    <ContentContainer className='content'>
      <ContentTitle>{title.toUpperCase()}</ContentTitle>
      <ContentSubtitle>SHOP NOW</ContentSubtitle>
    </ContentContainer>
  </MenuItemContainer>
)

export default withRouter(MenuItem)