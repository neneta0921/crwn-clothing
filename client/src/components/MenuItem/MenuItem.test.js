import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { MenuItem } from './MenuItem';

describe('MenuItem component', () => {
  let wrapper;
  let mockMatch;
  let mockHistory;
  const linkUrl = '/hats';
  const size = 'large';
  const imageUrl = 'testimage';

  beforeEach(() => {
    mockMatch = { url: './shop' };
    mockHistory = { push: jest.fn() };
    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      linkUrl,
      size,
      title: 'hats',
      imageUrl,
    };

    wrapper = shallow(<MenuItem {...mockProps} />);
  });

  it('should render MenuItem component', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call history.push with the right string whn MenuItemContainer clicked', () => {
    wrapper.find('MenuItemContainer').simulate('click');
    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}${linkUrl}`);
  });

  it('should pass size to MenuItemContainer as the prop size', () => {
    expect(wrapper.find('MenuItemContainer').prop('size')).toBe(size);
  });

  it('should pass imageUrl to BackgroundImageContainer as the prop imageUrl', () => {
    expect(wrapper.find('BackgroundImageContainer').prop('imageUrl')).toBe(imageUrl);
  });
});
