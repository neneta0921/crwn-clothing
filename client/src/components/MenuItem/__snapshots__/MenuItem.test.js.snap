// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component should render MenuItem component 1`] = `
<div
  className="sc-bdfBwQ iQFbfG"
  onClick={[Function]}
  size="large"
>
  <div
    className="sc-gsTCUz kZZfbi background-image"
  />
  <div
    className="sc-dlfnbm JasYd content"
  >
    <span
      className="sc-hKgILt eLkpOW"
    >
      HATS
    </span>
    <span
      className="sc-eCssSg cAcTCL"
    >
      SHOP NOW
    </span>
  </div>
</div>
`;
